robot_sdk/SDK/owl_sim_sdk/utils.py

```
@@ -0,0 +1,72 @@
+    # Convert quaternion to Euler angles (roll, pitch, yaw)
+    # Roll (x-axis rotation)
+    sinr_cosp = 2 * (qw * qx + qy * qz)
+    cosr_cosp = 1 - 2 * (qx * qx + qy * qy)
+    rx = math.atan2(sinr_cosp, cosr_cosp)
+    
+    # Pitch (y-axis rotation)
+    sinp = 2 * (qw * qy - qz * qx)
+    if abs(sinp) >= 1:
+        ry = math.copysign(math.pi / 2, sinp)  # use 90 degrees if out of range
+    else:
+        ry = math.asin(sinp)
```

Ruff (0.12.2)

Use ternary operator `ry = math.copysign(math.pi / 2, sinp) if abs(sinp) >= 1 else math.asin(sinp)` instead of `if`-`else`-block

(SIM108)